File: flutter_app/test/models/activity_entry_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:flutter_app/models/activity_entry.dart';

void main() {
  group('ActivityEntry', () {
    test('creates with required fields', () {
      final entry = ActivityEntry(
        id: '1', title: 'Med', time: '9 AM',
        type: ActivityType.medication, status: ActivityStatus.done,
        date: DateTime(2026, 6, 1),
      );
      expect(entry.id, '1');
      expect(entry.title, 'Med');
      expect(entry.type, ActivityType.medication);
      expect(entry.status, ActivityStatus.done);
    });

    test('toString returns readable string', () {
      final entry = ActivityEntry(
        id: '1', title: 'Med', time: '9 AM',
        type: ActivityType.medication, status: ActivityStatus.done,
        date: DateTime(2026, 6, 1),
      );
      expect(entry.toString(), 'ActivityEntry(Med, 9 AM, ActivityStatus.done)');
    });

    test('ActivityType enum has expected values', () {
      expect(ActivityType.values.length, 3);
      expect(ActivityType.values, contains(ActivityType.medication));
      expect(ActivityType.values, contains(ActivityType.appointment));
      expect(ActivityType.values, contains(ActivityType.alert));
    });

    test('ActivityStatus enum has expected values', () {
      expect(ActivityStatus.values.length, 4);
      expect(ActivityStatus.values, contains(ActivityStatus.done));
      expect(ActivityStatus.values, contains(ActivityStatus.missed));
      expect(ActivityStatus.values, contains(ActivityStatus.sent));
      expect(ActivityStatus.values, contains(ActivityStatus.todo));
    });
  });
}

File: react-native-app/src/models/__tests__/ActivityEntry.test.js
import { ActivityType, ActivityStatus, createActivityEntry } from '../ActivityEntry';

describe('ActivityEntry', () => {
  test('ActivityType has expected values', () => {
    expect(ActivityType.MEDICATION).toBe('medication');
    expect(ActivityType.APPOINTMENT).toBe('appointment');
    expect(ActivityType.ALERT).toBe('alert');
  });

  test('ActivityStatus has expected values', () => {
    expect(ActivityStatus.DONE).toBe('done');
    expect(ActivityStatus.MISSED).toBe('missed');
    expect(ActivityStatus.SENT).toBe('sent');
    expect(ActivityStatus.TODO).toBe('todo');
  });

  test('creates an activity entry', () => {
    const date = new Date();
    const entry = createActivityEntry({
      id: '1', title: 'Medicine', time: '9:00 AM',
      type: ActivityType.MEDICATION, status: ActivityStatus.DONE, date,
    });
    expect(entry.id).toBe('1');
    expect(entry.title).toBe('Medicine');
    expect(entry.type).toBe('medication');
    expect(entry.status).toBe('done');
    expect(entry.date).toBe(date);
  });
});
